cohort-engine/src/main/java/com/ibm/cohort/engine/CqlEngineWrapper.java

```diff
@@ -613,6 +622,17 @@ protected static Object resolveQuantityParameter(String value) {
 		typedValue = new Quantity().withValue(new BigDecimal(parts[0])).withUnit(parts[1]);
 		return typedValue;
 	}
+	
+	protected static void registerModelInfo(File modelInfoFile) {
+		ModelInfo modelInfo = JAXB.unmarshal(modelInfoFile, ModelInfo.class);
+		// Force mapping  to FHIR 4.0.1. Consider supporting different versions in the future.
+		// Possibly add support for auto-loading model info files.
+		modelInfo.setTargetVersion("4.0.1");
+		modelInfo.setTargetUrl("http://hl7.org/fhir");
```

Comment:
This feels wonky to me, but I get why you did it. It definitely simplifies the engine logic as we only need to support a data provider for the given FHIR URL and version and not every single custom IG under the sun. I was wondering if there is any chance, though, that this will conflict with the actual base FHIR modelinfo. I don't think so, but worth thinking about if you haven't already. Also, we just need to understand and agree that modelinfos for non-FHIR models are supported through this mechanism. I don't think that is a problem, but maybe something to put in the userguide.

Reply:
Best as I can tell, `targetUrl` and `targetVersion` are the mechanism that the CQF folks have in place to map back to base FHIR. Technically I am still waiting to hear back from them on what the intended behavior is. If we  need a target url and version set for the translator to  map to base  FHIR  properly, then I'd say we can either force one or just fail outright if the model info  does not already contain a target  version/url we support.

Note that `url` and `targetUrl` are two different fields in  the model info. `url` will still contain what is being mapped from (CDM url), and `targetUrl` will contain the FHIR url (what is being mapped  to). `name` and `version` should be what  is used to identify the model info itself. I don't think this should interfere with  the base FHIR modelinfo,  but  I can double check this.

Reply:
<img width="1051" alt="Screen Shot 2020-11-06 at 5 10 52 PM" src="https://user-images.githubusercontent.com/72105334/98419443-11ec3800-2053-11eb-845c-53b4fc308237.png">
Looks like once registered both base FHIR model info and the CDM model info are still available. They are keyed on name and version